File: src/main/java/com/github/sorhus/hmmongo/DNAEncoder.java
package com.github.sorhus.hmmongo;

import java.io.Serializable;

public class DNAEncoder implements Serializable {

    public static int encode(char ch) {
        if(ch == 'a' || ch == 'A') {
            return 0;
        } else if(ch == 'c' || ch == 'C') {
            return 1;
        } else if(ch == 'g' || ch == 'G') {
            return 2;
        } else if(ch == 't' || ch == 'T') {
            return 3;
        } else {
            throw new RuntimeException("Corrupt char: " + ch);
        }
    }
}

File: src/main/scala/com/github/sorhus/hmmongo/DNAEncoder.scala
package com.github.sorhus.hmmongo

import scala.util.{Failure, Success, Try}

class DNAEncoder(capitals: Boolean = false) extends Function[String, List[Array[Int]]] {

  override def apply(input: String): List[Array[Int]] = {
    Try(input.toCharArray.map(encode)) match {
      case Success(result) => result :: Nil
      case Failure(error) => Nil
    }
  }

  def encode(input: Char): Int = {
    val ch = if(capitals) input.toLower else input
    ch match {
      case 'a' => 0
      case 'c' => 1
      case 'g' => 2
      case 't' => 3
      case  _  => throw new IllegalArgumentException
    }
  }
}
